test/sql/misc.test.lua

```diff
@@ -79,3 +79,29 @@ box.execute('SELECT a;')
 diag = box.error.last()
 box.execute('SELECT * FROM (VALUES(true));')
 diag == box.error.last()
+
+-- exclude_null + SQL correctness
+box.execute([[CREATE TABLE j (s1 INT PRIMARY KEY, s2 STRING, s3 VARBINARY)]])
+s = box.space.J
+i = box.space.J:create_index('I3',{parts={2,'string', exclude_null=true}})
+box.execute([[INSERT INTO j VALUES (1,NULL,NULL), (2,'',X'00');]])
+
+box.execute([[SELECT * FROM j;]])
+box.execute([[SELECT * FROM j INDEXED BY I3;]])
+
+box.execute([[SELECT COUNT(*) FROM j GROUP BY s2;]])
+box.execute([[SELECT COUNT(*) FROM j INDEXED BY I3;]])
+
+box.execute([[UPDATE j INDEXED BY i3 SET s2 = NULL;]])
+box.execute([[INSERT INTO j VALUES (3, 'a', X'33');]])
+
+box.execute([[SELECT * FROM j;]])
+box.execute([[SELECT * FROM j INDEXED BY I3;]])
+
+box.execute([[UPDATE j INDEXED BY i3 SET s3 = NULL;]])
+s:select{}
+-- Fixme: exclude_null should be true
+box.space.J.index.I3:alter({parts={2,exclude_null=true}})
```

Comment:
Didn't get this test...Am I correct that you intended to shown that missing type specification in key definition results in re-setting exclude_null property to false? Can you fix it? Is it related to current PR (https://github.com/tarantool/tarantool/pull/5731)? If so, could you please move this case out of sql suite?

Reply:
Yes? it's about this bug. I'll remove this test case from this pull request then.